File: day-02/main.rb
# Advent of Code 2015 - Day 2

def read_input
  File.read(File.expand_path("../input.txt", __FILE__))
end

def one_paper(l, w, h)
  area = 2 * (l * w + l * h + w * h)
  slack = l * w # relies on [l, w, h] being sorted
  area + slack
end

def one_ribbon(l, w, h)
  2 * (l + w) + l * w * h # relies on [l, w, h] being sorted
end

def main
  input = read_input.split("\n").reject(&:empty?)

  paper = 0
  ribbon = 0

  input.each do |line|
    dims = line.split("x").map(&:to_i).sort
    paper += one_paper(*dims)
    ribbon += one_ribbon(*dims)
  end

  puts "Part One: #{paper}" # square feet of wrapping paper
  puts "Part Two: #{ribbon}" # feet of ribbon
end

main if "main" == to_s

File: day-02/main.py
"""Advent of Code 2015 - Day 2."""

from pathlib import Path


def read_input():
    """Read input file and split into individual lines returned as a list."""
    return Path(__file__).with_name('input.txt').read_text().splitlines()


def one_paper(l, w, h):
    """Compute needed wrapping paper for one present.

    Arguments l, w, and h are expected to be sorted for a correct result.
    """
    area = 2 * (l * w + l * h + w * h)
    slack = l * w
    return area + slack


def one_ribbon(l, w, h):
    """Compute needed ribbon for one present.

    Arguments l, w, and h are expected to be sorted for a correct result.
    """
    return 2 * (l + w) + l * w * h


def main():
    """Main entry point of puzzle solution."""
    lines = read_input()

    paper = 0
    ribbon = 0

    for line in lines:
        dims = sorted(map(int, line.split('x')))
        paper += one_paper(*dims)
        ribbon += one_ribbon(*dims)

    print('Part One: {}'.format(paper))
    print('Part Two: {}'.format(ribbon))


if __name__ == '__main__':
    main()
